akka-stream/src/main/scala/akka/stream/impl/fusing/Ops.scala

```diff
@@ -1328,56 +1329,66 @@ final class Delay[T](val d: FiniteDuration, val strategy: DelayOverflowStrategy)
           }
```

Comment:
I can do the above refactoring also, but probably easier to review this bug fix first.
`FIXME rewrite into distinct strategy functions to avoid matching on strategy for every input when full`